dragonfly/implementations/psyllid_provider.py

```diff
@@ -233,51 +238,167 @@ def get_number_of_streams(self, channel):
+        self.provider.set(self.queue_dict[channel]+request, threshold)
+        logger.info('Setting psyllid power snr threshold to {}'.format(threshold))
+
+
+    def _get_fmt_snr_threshold(self, channel='a'):
+        request = '.active-config.{}.fmt.threshold-power-snr'.format(str(self.channel_dict[channel]))
+        threshold = self.provider.get(self.queue_dict[channel]+request)['threshold-power-snr']
+        return float(threshold)
+
+
+    def _set_fmt_snr_high_threshold(self, threshold, channel='a'):
+        request = '.active-config.{}.fmt.threshold-power-snr-high'.format(str(self.channel_dict[channel]))
+        self.provider.set(self.queue_dict[channel]+request, threshold)
+        logger.info('Setting psyllid power snr threshold to {}'.format(threshold))
+
+
+    def _get_fmt_snr_high_threshold(self, channel='a'):
+        request = '.active-config.{}.fmt.threshold-power-snr-high'.format(str(self.channel_dict[channel]))
+        threshold = self.provider.get(self.queue_dict[channel]+request)['threshold-power-snr-high']
+        return float(threshold)
+
+
+    def _set_trigger_mode(self, mode_id, channel='a'):
+        request = '.active-config.{}.fmt.trigger-mode'.format(str(self.channel_dict[channel]))
+        self.provider.set(self.queue_dict[channel]+request, mode_id)
+        logger.info('Setting psyllid trigger mode to {} threshold trigger'.format(mode_id))
+
+
+    def _get_trigger_mode(self, channel='a'):
+        request = '.active-config.{}.fmt.trigger-mode'.format(str(self.channel_dict[channel]))
+        trigger_mode = self.provider.get(self.queue_dict[channel]+request)['trigger-mode']
+        return trigger_mode
+
+
+    def _set_n_triggers(self, n_triggers, channel='a'):
+        request = '.active-config.{}.eb.n-triggers'.format(str(self.channel_dict[channel]))
+        self.provider.set(self.queue_dict[channel]+request, n_triggers)
+        logger.info('Setting psyllid n-trigger/skip-tolerance to {}'.format(n_triggers))
+
+
+    def _get_n_triggers(self, channel='a'):
+        request = '.active-config.{}.eb.n-triggers'.format(str(self.channel_dict[channel]))
+        n_triggers = self.provider.get(self.queue_dict[channel]+request)['n-triggers']
+        return n_triggers
+
+
+    # tell psyllid to record a frequency mask, write it to a json file and prepare for triggered run
+    def make_trigger_mask(self, channel='a', filename='~/fmt_mask.json'):
+        if self.mode_dict[channel] != 'triggered':
+            logger.error('Psyllid instance is not in triggered mode')
+            raise core.exceptions.DriplineGenericDAQError("Psyllid instance is not in triggered mode")
 
-        logger.info('freq-only')
-        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.freq-only'
+        logger.info('Switch tf_roach_receiver to freq-only')
+        request = 'run-daq-cmd.{}.tfrr.freq-only'.format(str(self.channel_dict[channel]))
         result = self.provider.cmd(self.queue_dict[channel],request)
```

Comment:
if there is an error in the procedure psyllid will tell you

Reply:
If the result isn't being used, you can replace `result = self.provider.cmd(<stuff>)` with `self.provider.cmd(<stuff>)` and let the result be garbage collected before placing it into the local scope.

When you say "psyllid will tell you" do you mean it will send a message for the user? If psyllid sends a reply with a non-zero error code to one of the first cmd requests, doesn't that mean that there is an error and making the mask has failed? This should raise an exception here and return an error code to whatever requested it of this service.

Reply:
Actually that's not even true. The implementation of `provider.cmd` should already be checking the return code and raising the exception. you should just not catch the `result` into a local unless the cmd produces some useful content in the payload which you need to use (which doesn't appear to be the case here anywhere).